transports/tcp/src/lib.rs

```diff
@@ -734,116 +721,88 @@ where
+            // If the listener is bound to a single interface, make sure the
+            // address is registered for port reuse and reported once.
+            InAddr::One(out) => {
+                if let Some(multiaddr) = out.take() {
+                    me.port_reuse
+                        .register(me.listen_addr.ip(), me.listen_addr.port());
+                    return Poll::Ready(Some(Ok(TcpListenerEvent::NewAddress(multiaddr))));
                 }
             }
+        }
 
-            // Take the pending connection from the backlog.
-            let incoming = match T::poll_accept(&mut me.listener, cx) {
-                Poll::Pending => return Poll::Pending,
-                Poll::Ready(Ok(incoming)) => incoming,
-                Poll::Ready(Err(e)) => {
-                    // These errors are non-fatal for the listener stream.
-                    log::error!("error accepting incoming connection: {}", e);
-                    me.pause = Some(Delay::new(me.sleep_on_error));
-                    return Poll::Ready(Some(Ok(TcpListenerEvent::Error(e))));
+        if let Some(mut pause) = me.pause.take() {
+            match Pin::new(&mut pause).poll(cx) {
+                Poll::Ready(_) => {}
+                Poll::Pending => {
+                    me.pause = Some(pause);
+                    return Poll::Pending;
                 }
-            };
+            }
+        }
 
-            let local_addr = ip_to_multiaddr(incoming.local_addr.ip(), incoming.local_addr.port());
-            let remote_addr =
-                ip_to_multiaddr(incoming.remote_addr.ip(), incoming.remote_addr.port());
+        // Take the pending connection from the backlog.
+        let incoming = match T::poll_accept(&mut me.listener, cx) {
+            Poll::Pending => return Poll::Pending,
+            Poll::Ready(Ok(incoming)) => incoming,
+            Poll::Ready(Err(e)) => {
+                // These errors are non-fatal for the listener stream.
+                log::error!("error accepting incoming connection: {}", e);
+                me.pause = Some(Delay::new(me.sleep_on_error));
+                return Poll::Ready(Some(Ok(TcpListenerEvent::Error(e))));
+            }
+        };
 
-            log::debug!("Incoming connection from {} at {}", remote_addr, local_addr);
+        let local_addr = ip_to_multiaddr(incoming.local_addr.ip(), incoming.local_addr.port());
+        let remote_addr = ip_to_multiaddr(incoming.remote_addr.ip(), incoming.remote_addr.port());
 
-            return Poll::Ready(Some(Ok(TcpListenerEvent::Upgrade {
-                upgrade: future::ok(incoming.stream),
-                local_addr,
-                remote_addr,
-            })));
-        }
+        log::debug!("Incoming connection from {} at {}", remote_addr, local_addr);
+
+        Poll::Ready(Some(Ok(TcpListenerEvent::Upgrade {
+            upgrade: future::ok(incoming.stream),
+            local_addr,
+            remote_addr,
+        })))
```

Comment:
This is a bit of stylistic break from how we usually do things.

In `poll` functions with multiple sub futures, we tend to return `Poll::Pending` at the bottom and return early with `Poll::Ready` from `match` blocks.

This way, all sections of the the `poll` functions look the same.

Reply:
Fixed with 8d16ad7. Was that what you meant? :)